crates/artifacts/resolc/src/contract.rs

```diff
@@ -0,0 +1,161 @@
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{ResolcEVM, EVM};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ResolcContract {
+    /// The contract ABI.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub abi: Option<JsonAbi>,
+    /// The contract metadata.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<serde_json::Value>,
+    /// The contract developer documentation.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub devdoc: Option<DevDoc>,
+    /// The contract user documentation.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub userdoc: Option<UserDoc>,
+    /// The contract storage layout.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub storage_layout: Option<StorageLayout>,
+    /// Contract's bytecode and related objects
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub evm: Option<EVM>,
+    /// Revive related output
+    /// We are going to use  structs defined locally
+    /// as opposed to revive defined
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub resolc_evm: Option<ResolcEVM>,
+    /// The contract optimized IR code.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub ir_optimized: Option<String>,
+    /// The contract PolkaVM bytecode hash.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub hash: Option<String>,
+    /// The contract factory dependencies.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub factory_dependencies: Option<BTreeMap<String, String>>,
+    /// The contract missing libraries.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub missing_libraries: Option<HashSet<String>>,
+}
+
+impl Default for ResolcContract {
+    fn default() -> Self {
+        Self {
+            abi: None,
+            metadata: None,
+            devdoc: None,
+            userdoc: None,
+            storage_layout: None,
+            evm: None,
+            ir_optimized: None,
+            hash: None,
+            factory_dependencies: None,
+            missing_libraries: None,
+            resolc_evm: None,
+        }
+    }
+}
+
+impl<'a> From<&'a ResolcContract> for CompactContractBytecodeCow<'a> {
+    fn from(value: &'a ResolcContract) -> Self {
+        if let Some((standard_abi, compact_bytecode, compact_deployed_bytecode)) =
+            create_compact_bytecode(value)
+        {
+            Self {
+                abi: Some(Cow::Owned(standard_abi)),
+                bytecode: Some(Cow::Owned(compact_bytecode)),
+                deployed_bytecode: Some(Cow::Owned(compact_deployed_bytecode)),
+            }
+        } else {
+            Self { abi: None, bytecode: None, deployed_bytecode: None }
+        }
+    }
+}
+
+impl From<ResolcContract> for CompactContractBytecode {
```

Comment:
```suggestion
impl From<ResolcContract> for CompactContractBytecode {
    fn from(value: ResolcContract) -> Self {
        Self {
            abi: value.json_abi(),
            bytecode: value.compact_bytecode(),
            deployed_bytecode: value.compact_deployed_bytecode(),
        }
    }
}
```
or it's intended to always return every field to None if at least one of them is None?